tensorboard/plugins/hparams/api.proto

```diff
@@ -0,0 +1,187 @@
+  // the experiment. Currently, the UI considers hyperparameter values as opaque
+  // strings.
+  repeated string vals = 4;
+}
+
+message MetricInfo {
+  // A name for the MetricInfo. Within an experiment there should be at most
+  // one MetricInfo having the given name.
+  string name = 1;
+
+  // A string used to display the hyperparameter in the UI. If empty the UI
+  // will display the 'name' field.
+  string display_name = 2;
+
+  string description = 3;
+
+  // The dataset type (validation, training) on which the metric is computed.
+  DatasetType dataset_type = 4;
+}
+
+message SessionGroup {
+  string name = 1;
+  repeated HParamVal hparam_vals = 2;
+}
+
+message HParamVal {
+  string name = 1;
+  string value = 2;
+}
+
+message Session {
+  // An id for the session. Unique within an experiment.
+  string name = 1;
+
+  google.protobuf.Timestamp time_started = 2;
+
+  // May be empty if unavailable or the session has not finished yet.
+  google.protobuf.Timestamp time_completed = 3;
+
+  // May be STATUS_UNKNOWN if unavailable.
+  Status status = 4;
+
+  // A filesystem path that will allow the user to reconstruct the model for
+  // this session. E.g., in Tensorflow this should be the directory where
+  // checkpoints are saved.
+  string model_path = 5;
+
+  // An optional link to a website monitoring the session.
+  string monitor_url = 6;
+}
+
+enum DatasetType {
+  DATASET_UNKNOWN = 0;
+  DATASET_TRAINING = 1;
+  DATASET_VALIDATION = 2;
+}
+
+enum Status {
+  STATUS_UNKNOWN = 0;
+  STATUS_SUCCESS = 1;
+  STATUS_FAILURE = 2;
+  STATUS_RUNNING = 3;
+}
+
+// We document the API below. Each end-point below is assumed to be called
+// vi an HTTP GET and arguments encoded in the URL . Additionally, each
+// end-point starts with 'v1' to support multiple versions.
+// Note that per Tensorboard's conventions the end-point would be
+// prefixed by "data/plugin/hparams" (e.g. the get_experiment end-point is
+// actually "data/plugin/hparams/v1/get_experiment").
+//
+// HTTP end-point:
+//   v1/get_experiment
+// Args:
+//   None.
+// Description:
+//   Returns an Experiment object defining metadata for the experiment.
+//
+// HTTP end-point
+//   v1/list_session_groups
+// Args:
+//   None.
+// Returns:
+//   An array of SessionGroup objects.
+//
+// HTTP end-point:
+//   v1/list_sessions
+// Args:
+//   session_group: The name of the session group to list.
+// Returns:
+//   An array of Session objects.
+//
+// HTTP end-point:
+//   v1/list_metric_evals
+// Args:
+//   session_group, session, metric: Names for the session_group, session and
+//   metric that identify the metric-data to return.
+// Returns:
+//   An array consisting of evaluations of the metric identified by 'metric'
+//   in the session identified by 'session' within the session group
+//   identified by 'session_group'. Each element in the array is itself
+//   a 3-element array of the form [wall_time, step, value], where:
```

Comment:
For [wall_time, step, value]
are both wall_time and step required?
In our case, we only force users to report one of them, so we'll likely cannot populate both fields.

Reply:
+1 to Nick's question. Sagi, can you answer?